lib/matplotlib/tri/tricontour.py

```diff
@@ -84,68 +84,31 @@ def _contour_args(self, args, kwargs):
 _docstring.interpd.update(_tricontour_doc="""
 Draw contour %(type)s on an unstructured triangular grid.
 
-The triangulation can be specified in one of two ways; either ::
+Call signatures::
 
-    %(func)s(triangulation, ...)
+    %(func)s(triangulation, Z, [levels], ...)
+    %(func)s(x, y, Z, [levels], *, [triangles=triangles], [mask=mask], ...)
 
-where *triangulation* is a `.Triangulation` object, or ::
+The triangular grid can be specified either by passing a `.Triangulation`
+object as the first parameter, or by passing the points *x*, *y* and
+optionally the *triangles* and a *mask*. See `.Triangulation` for an
+explanation of these parameters. If neither of *triangulation* or
+*triangles* are given, the triangulation is calculated on the fly.
 
-    %(func)s(x, y, ...)
-    %(func)s(x, y, triangles, ...)
-    %(func)s(x, y, triangles=triangles, ...)
-    %(func)s(x, y, mask=mask, ...)
-    %(func)s(x, y, triangles, mask=mask, ...)
-
-in which case a `.Triangulation` object will be created.  See that class'
-docstring for an explanation of these cases.
-
-The remaining arguments may be::
-
-    %(func)s(..., Z)
-
-where *Z* is the array of values to contour, one per point in the
-triangulation.  The level values are chosen automatically.
-
-::
-
-    %(func)s(..., Z, levels)
-
-contour up to *levels+1* automatically chosen contour levels (*levels*
-intervals).
-
-::
-
-    %(func)s(..., Z, levels)
-
-draw contour %(type)s at the values specified in sequence *levels*, which must
-be in increasing order.
-
-::
-
-    %(func)s(Z, **kwargs)
-
-Use keyword arguments to control colors, linewidth, origin, cmap ... see below
-for more details.
+It is possible to pass *triangles* positionally, i.e.
```

Comment:
In the `tripcolor` docs this is in a separate Notes section. Do you want to do the same here? On the other hand, if it is discouraged, do you even want to mention it as a possibility?

Reply:
Since `x, y, triangles, Z` was a previously documented variant, I don't want to silently undocument it. We either have to stick with the discouraged notice or should go with a proper deprecation. It's not so easy to add notes here, because the docstring is left partial so that `tricontour`/ `tricontourf` are able to add additional specific parameters. In general, I think this information is better placed near the call signature. I've moved the info up in `tripcolor` so that they are consistent.